aten/src/ATen/native/cuda/Reduce.cuh

```diff
@@ -473,17 +486,48 @@ struct ReduceOp {
     return arg_t {};
   }
 
+  // This function should never be called --
+  // it's the version of `get_accumulated_output`
+  // when accumulation in the output is not possible.
   template <bool can_acc>
-  C10_DEVICE out_scalar_t project_if_necessary(
-    arg_t value,
+  C10_DEVICE out_scalar_t get_accumulated_output(
+    out_scalar_t* out, arg_t value,
     typename std::enable_if<!can_acc>::type* = nullptr
   ) const {
+    assert(false);
+    return *out;
+  }
+
+  template<class T>
+  C10_DEVICE void set_results(const T x, const index_t base_offset) const {
+    assert(noutputs == 1);
+    auto res = (out_scalar_t*)((char*)dst[0] + base_offset);
+    *res = x;
+  }
+
+  //Currently implemented for max of two outputs
+  template<class T>
+  C10_DEVICE void set_results(const thrust::tuple<T, T> x, const index_t base_offset) const {
+    if (noutputs >= 1) {
```

Comment:
How could these ever be false? Won't `noutputs` always be exactly two if we get here? (Unless there is some bug)

Reply:
We share WelfordOps between std and std_mean. First function will have noutputs = 1 and second noutputs = 2, even if the result will always contain two values.